### ISSUE TITLE
[QUESTION] Handle re-connect to server?

### ISSUE THREAD

**Issue by ValiDrv** [open-smf/connection-pool#7]
Can this connection pool handle re-connecting

Ex:
```
- Connection to the mysql server established
- Commands can be sent correctly
- Network cable is unplugged/disconnected.
- Commands try to send, detect that the network disconnected, and tries to reconnect.
- Network cable is plugged back in/network goes up
- Pool eventually reconnects, and resumes operations correctly.
```

**Comment by hhxsv5** [comment]
Yes. The connection status will be checked when borrowing, and the disconnected connection will be removed from connection pool. [See here](https://github.com/open-smf/connection-pool/blob/master/src/ConnectionPool.php#L133).

**Comment by ValiDrv** [comment]
Are you sure that works when the network drops, and not just if the socket is connected? ex: swoole_event_addConnected

**Comment by kaerusdev** [comment]
This condition, which returns an throw, prevents the code from being executed further, and therefore the connection is not re-created.

https://github.com/open-smf/connection-pool/blob/f9289cb5ee61d3e901bc74ab745e5b2162461a1e/src/ConnectionPool.php#L131
